packages/consumption/src/modules/openid4vc/OpenId4VcController.ts

```diff
@@ -1,34 +1,53 @@
-import { ClaimFormat, W3cJsonCredential } from "@credo-ts/core";
-import { OpenId4VciCredentialResponse, OpenId4VciResolvedCredentialOffer, OpenId4VpResolvedAuthorizationRequest } from "@credo-ts/openid4vc";
+import { ClaimFormat } from "@credo-ts/core";
+import { OpenId4VciResolvedCredentialOffer, OpenId4VpResolvedAuthorizationRequest } from "@credo-ts/openid4vc";
 import { VerifiableCredential } from "@nmshd/content";
 import { ConsumptionBaseController } from "../../consumption/ConsumptionBaseController";
 import { ConsumptionController } from "../../consumption/ConsumptionController";
 import { ConsumptionControllerName } from "../../consumption/ConsumptionControllerName";
 import { OwnIdentityAttribute } from "../attributes";
 import { Holder } from "./local/Holder";
 import { KeyStorage } from "./local/KeyStorage";
+import { OpenId4VciCredentialResponseJSON } from "./local/OpenId4VciCredentialResponseJSON";
+import { RequestedCredentialCache } from "./local/RequestedCredentialCache";
 
 export class OpenId4VcController extends ConsumptionBaseController {
     private holder: Holder;
+    private credentialRequestCache: RequestedCredentialCache;
 
     public constructor(parent: ConsumptionController) {
         super(ConsumptionControllerName.OpenId4VcController, parent);
     }
 
     public override async init(): Promise<this> {
-        const collection = await this.parent.accountController.getSynchronizedCollection("openid4vc-keys");
-        const keyStorage = new KeyStorage(collection, this._log);
+        const keyCollection = await this.parent.accountController.getSynchronizedCollection("openid4vc-keys");
+        const keyStorage = new KeyStorage(keyCollection, this._log);
 
         this.holder = new Holder(keyStorage, this.parent.accountController, this.parent.attributes, this.fetchInstance);
         await this.holder.initializeAgent("96213c3d7fc8d4d6754c7a0fd969598e");
 
+        const requestedCredentialsCacheCollection = await this.parent.accountController.getSynchronizedCollection("openid4vc-requested-credentials-cache");
+        this.credentialRequestCache = new RequestedCredentialCache(requestedCredentialsCacheCollection);
+
         return this;
     }
 
     private get fetchInstance(): typeof fetch {
         return this.parent.consumptionConfig.fetchInstance ?? fetch;
     }
 
+    public async requestCredentialsCached(credentialOfferUrl: string): Promise<OpenId4VciCredentialResponseJSON[]> {
```

Comment:
I'm not a big fan of this method naming, but I don't have a better idea.. I'm open for suggestions.

Reply:
Is it even necessary to let the outside know that it checks a cache?

Reply:
I think intuitively I only would have provided `resolveCredentialOffer`, which internall does the caching.